### DISCUSSION TITLE
v9.0.0

### DISCUSSION THREAD

**Discussion by tadashi-aikawa** [tadashi-aikawa/obsidian-various-complements-plugin#206] [Announcements]
### Full customizable hotkeys (#245) `🏷️beta1`

You can now set any number of your favorite hotkeys on the popup!!🥳 

![image](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/assets/9500018/8bbae6a8-771e-4f42-820f-a29f83ee7a14)

Instead, I removed some options that would be unnecessary. Please migrate your settings with reference to "🔥 Breaking changes" sections.

#### Related issues.

- [x] #138  `🏷️beta1`
- [x] #156  `🏷️beta1`
- [x] #176  `🏷️beta1`

## ✨ **New things**

- [x] Add support for many quotation patterns around the world (#170) `🏷️beta2`
- [x] Add an "insert as text" command in the popup (#240) `🏷️beta2` (Fix bugs in `🏷️beta3`)
- [x] Add "select 1st" ... "select 9th" commands of the popup (#141) `🏷️beta4`
- [x] Add a "Enable Intelligent Suggestion Prioritization" option (#234) `🏷️beta5`

## 🔼 **Improvements**

* Surround a suggestion with double quotes if the suggestion of front matters is an internal link (#250) `🏷️beta5`
* The "Front matter complements" can now propose suggestions in edge cases more than before (#248) `🏷️beta6`

### Remove following options `🏷️beta1`

- [⚙️ Select a suggestion key](https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/4.%20Options/4.3.%20Key%20customization/%E2%9A%99%EF%B8%8F%20Select%20a%20suggestion%20key/)
    - Set "select" in the Key customization instead of the option
- [⚙️ Additional cycle through suggestions keys](https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/4.%20Options/4.3.%20Key%20customization/%E2%9A%99%EF%B8%8F%20Additional%20cycle%20through%20suggestions%20keys/)
    - Set "up" and "down" in the Key customization instead of the option
- [⚙️ Disable the up,down keys for cycle through suggestions keys](https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/4.%20Options/4.3.%20Key%20customization/%E2%9A%99%EF%B8%8F%20Disable%20the%20up%2Cdown%20keys%20for%20cycle%20through%20suggestions%20keys/) 
    - Remove "ArrowDown" from "up", and "ArrowUp" from "down", in the Key customization instead of the option
- [⚙️ Open source file key](https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/4.%20Options/4.3.%20Key%20customization/%E2%9A%99%EF%B8%8F%20Open%20source%20file%20key/)
    - Set "open" in the Key customization instead of the option
- (Experimental) Use common prefix completion of suggestion
    - Set "completion" in the Key customization instead of the option

**Comment by brimwats** [comment]
this looks fantastic and a great compromise!

**Reply by tadashi-aikawa** [reply]
Thanks!

In this release, I had been refactoring codes related to hotkeys and commands. So it will be easier to implement new features as a command than past version 👍
